File: Stirling'S Approximation/stirling.c
#include "stdio.h"
#include "math.h"

int n;
double atas = 0;
double bawah;
double all;
double out;
int c;

int main(){
    while (!feof(stdin)) {
        scanf("%d", &n);
        if (n == 0){
            break;
        }
        for (c = 1; c <= n; c++){
            atas = atas + log(c);
        }
        out = pow(2.71828182, atas-(n+0.5)*log(n)+n-0.918938533);
        printf("%8.8f\n", out);
        atas = 0;
    }
}

File: Stirling'S Approximation/stirling.py
import sys
import math

for line in sys.stdin:
	if line == "0\n":
		break
	n = int(line)
	out = 0
	for i in range (1, n+1):
		out = out + math.log(i)
	print 2.71828182**(out-(n+0.5)*math.log(n)+n-0.918938533)